추가정보 페이지 추가 및 리팩토링

jma1020 opened the pull request:
# Pull Request
- [x] feature

## Summary
추가정보 페이지 구현 및 리팩토링

## Describe your changes
- name, 성별, 생년월일, 지역구 유효성 검사 적용
- 날짜Data를 가져오는 list 함수
- Select 컴포넌트 생성
- 지역구 Data 커스텀 hook 작성

## Issue number and link
- #7 

## ETC
- Select 에서 option들을 children이 아니라 Select 내에서 처리할 수 있도록 리팩토링 예정
- fetch hook은 추후 useSWR, React-query, RTK-query로 마이그레이션 예정
- 지역 code 프로퍼티를 가져오는 부분 리팩토링 예정

jma1020 committed:
feat: date 함수 구현

- year 데이터  구현
- month 데이터 구현
- day 데이터 구현

jma1020 committed:
메세지수정

jma1020 committed:
feat:추가정보 페이지  schema추가

- 추가정보 페이지에 필요한 schema들 구현
- 추가정보 schema 타입 지정

jma1020 committed:
feat: 추가정보 페이지 구현 및 react-hook-form 적용

- 전체적인 react-hook-form 적용
- 받아온 생년월일에 맞는 select option 구현
- 시 군 구 data api 를 이용해서 select 박스 틀 구축

jma1020 committed:
feat: 추가정보 페이지 Select 컴포넌트화 작업 및 생년월일 리팩토링

- 기존 state -> watch로 변경
- Select 태그 컴포넌트화 작업

jma1020 committed:
단순 오타 수정

jma1020 committed:
불필요한 주석 제거

jma1020 committed:
feat: 지역 데이터 fetch 커스텀 hook 작성

- 도시를 가져오는 커스텀 useCity hook
- 구 지역을 가져오는 커스텀 useDistrict hook

jma1020 committed:
feat: 지역 데이터를 가져오는 hook 하나로 통일(리팩토링)

- data fetch 커스텀 hook 리팩토링
- 새로만든 hook 적용

jma1020 committed:
console 제거

jma1020 committed:
불필요한 주석 삭제

jma1020 committed:
feat: Select 컴포넌트 export, styled  파일 추가

- styled 지정 파일 별도로 생성
- export 파일 별도로 생성

jma1020 committed:
feat: Select 컴포넌트 리팩토링

- 각 type 지정
- children으로 option data 를 보여주는 구조 삭제
- Select 컴포넌트안에서 형식에 맞춰서 option 생성
- default option 생성
- styled 파일에서 styled컴포넌트 사용

jma1020 committed:
feat: Date 파일 date return 형식 변경

- 변경한 Select 컴포넌트 형식에 맞게끔 date return 형식 변경

jma1020 committed:
feat: useRegionFetch 리팩토링

- type 지정
- 행정구역 data에 code와 name으로 이루어진 Map 객체 생성 (추후 backend로
  submit 시 활용)
- Select 컴포넌트 형식에 맞춰서 data 구조 변경

jma1020 committed:
feat: 추가정보 페이지 리팩토링

- 기존 state -> watch로 변경
- 리팩토링한 Select 컴포넌트 적용

jma1020 committed:
feat: schema 벨리데이션 추가

- Select에서 사용하는 defaultValue 예외처리

jma1020 committed:
사용하지 않는 파일 삭제

jma1020 committed:
🔥 주석 삭제

jma1020 committed:
변수이름 변경 및 주석 삭제

jma1020 committed:
Merge branch 'develop' into feature/add-information

jma1020 committed:
Merge branch 'feature/add-information' of https://github.com/TRIPONG-2022/frontend into feature/add-information

jma1020 commented:
충돌난거 해결하다가 저도 모르게 충돌된 부분 Merge가 된 것 같습니다ㅠㅠ
번거롭겠지만 한번 확인 부탁드립니닷!!..

sangrimlee commented on components/shared/Select/Select.tsx:
```diff
@@ -0,0 +1,40 @@
+
+          {options?.map(({ value, label }) => (
+            <Styled.Option key={`${id}-${value}`} value={value}>
+              {typeof label == 'string' && label.split(' ')[1]
```
`useRegionFetch`에서 `SelectOption`형태로 변경할 때 `split`해서, `Select` Component 내에서 `split`하지않는게 어떨까요?

왜냐하면 `Select` Component는 나중에 또 다른데서 쓰이는데, `useRegionFetch`는 다른 곳에서 사용되지는 않으니깐요:)

jma1020 replied:
감사합니다!! 코멘트 주신 부분 적용했어요 ㅎㅎ

sangrimlee commented on constants/schema.ts:
```diff
@@ -32,6 +32,35 @@ export const LOGIN_SCHEMA = yup.object({
+    .required(SCHEMA_MESSAGES.REQUIRED_FIELD),
+});
+
+export type InformationScheme = yup.InferType<typeof ADD_INFORMATION_SCHEMA>;
```
InformationScheme을 InformationSchema로 바꾸셔야 될 것 같아요😄 

다른 곳들에서 개인적인 주석들은 지워주시면 좋을 것 같아요!

jma1020 replied:
앗.. 오타까지 찾아주시다니 너무 감사합니다🥰

seongjin2427 reviewed (approved):
확인했습니다! 😀

jma1020 committed:
feat: 코드 리뷰를 통한 Select함수 공용화 피드백 적용

- 기존 Select에 split으로 값을 가져오던 것 삭제
- useRegionFetch hook 에서 지역'구' 에 들어갈 value 조정

jma1020 committed:
🩹 Schema 오타 수정

sangrimlee reviewed (approved):
주석 부분만 지우시면 될 것 같아요! 😃

jma1020 committed:
🔥 불필요한 주석삭제
